packages/mgt-components/src/components/mgt-todo/mgt-todo.ts

```diff
@@ -572,12 +641,35 @@ export class MgtTodo extends MgtTasksBase {
     }
   };
 
+  private readonly handleChange = async (e: KeyboardEvent, task: TodoTask) => {
+    if ((e.target as HTMLInputElement).id === task.id) {
+      if (e.key === 'Enter') {
+        await this.updateTask(task);
+        (e.target as HTMLInputElement)?.blur();
+      }
+      this._changedTaskName = (e.target as HTMLInputElement).value;
+    }
+  };
+
   private readonly handleKeyDown = async (e: KeyboardEvent) => {
-    if (e.key === 'Enter') {
+    if (e.key === 'Enter' && (e.target as HTMLInputElement).id === 'new-task-name-input') {
       await this.addTask();
     }
   };
 
+  private readonly updatingTask = (e: KeyboardEvent, task: TodoTask) => {
+    if ((e.target as HTMLInputElement).id === task.id) {
+      this._taskBeingUpdated = task;
+    }
+  };
+
+  private readonly handleBlur = async (e: Event, task: TodoTask) => {
+    if ((e.target as HTMLInputElement).id === task.id) {
+      await this.updateTask(task);
+      (e.target as HTMLInputElement)?.blur();
```

Comment:
There's a strange interaction when using the keyboard as your input device where pressing tab with the title/name text input focused where the focus moves to the date input correctly and then the date input becomes invisible.

I suspect that the updateTask promise is completing in the background and something happens to cause the date input to lose focus, either due to the blur method being called here or as a side effect of updateTask.

Perhaps we need a common parent to the two inputs that the blur event listener is attached to?

Reply:
Thank you! Missed this. Yes, I've implemented a common parent for both fluent-text-fields

Reply:
In testing it looks like the blur of the common parent isn't firing, possibly because the element itself is not interactive, or possibly becuase blur doesn't bubble while the lostfocus does.

this stuff is tricky to get right but it's important to make sure this works well. 

I'd suggest that you try the lostfocus event first to see if that fires when we want it to. if not then maybe we need to consider a refactor to look at using document.activeElement to see if either of the text inputs for the row are the active element before sending the update request.

Reply:
@Mnickii incase you missed my reply here.